extensions/iframe/common/client.mjs

```diff
@@ -0,0 +1,73 @@
+// Copyright 2021 HITCON Online Contributors
+// SPDX-License-Identifier: BSD-2-Clause
+
+import Overlay from '/static/sites/game-client/ui/overlay.mjs';
+import Modal from '/static/sites/game-client/ui/modal.mjs';
+import OverlayPosition from '/static/sites/game-client/ui/overlay-position.mjs';
+import {MapCoord} from '/static/common/maplib/map.mjs';
+
+class IframeOverlay extends Overlay {
+  constructor(mainUI) {
+    const dom = document.getElementById('iframe-overlay');
+    super(mainUI, dom);
+    this.iframe = dom.querySelector('iframe');
+    this.hide();
+    this.isOpen = false;
+  }
+
+  display(src, attributes = {}) {
+    if (this.isOpen) return;
+    this.iframe.src = src;
+    this.show(OverlayPosition.LEFT_BOTTOM);
+    this.isOpen = true;
```

Comment:
Maybe implement onPostShow() so that even if someone else called show(), isOpen will be correct.

Reply:
Try to get this done....

Reply:
I realized that I forget about Class can do, thx